pylearn2/utils/data_specs.py

```diff
@@ -265,7 +266,7 @@ def nest(self, flat):
                 assert self.n_unique_specs == 1
             return self._make_nested_space(flat, self.spec_mapping)
         else:
-            if isinstance(flat, tuple):
+            if isinstance(flat, Sized):
```

Comment:
The problem with "Sized" is that it would match a string, and try to match the length of the string with the number of elements needed, whereas in that case we want `n_unique_specs` to be 1, and the whole string to be nested.